Added important note to solve #3421 (#3758)
MicrosoftDocs/windows-itpro-docs: windows/security/identity-protection/hello-for-business/hello-hybrid-cert-trust-devreg.md

> [!IMPORTANT]
> If your environment is not federated, review the [New Installation baseline](hello-hybrid-cert-new-install.md) section of this deployment document to learn how to federate your environment for your Windows Hello for Business deployment. 

> >[!IMPORTANT]

Reviewer: ```suggestion
```

> >[!IMPORTANT]

Reviewer: ```suggestion
>[!TIP]
```

> > The identity team has simpolified the join flow for Azure AD device registration, the tutorial can be found [here](https://docs.microsoft.com/azure/active-directory/devices/hybrid-azuread-join-federated-domains).

Reviewer: ```suggestion
```

> > The identity team has simpolified the join flow for Azure AD device registration, the tutorial can be found [here](https://docs.microsoft.com/azure/active-directory/devices/hybrid-azuread-join-federated-domains).

Reviewer: ```suggestion
>Refer to the [Tutorial: Configure hybrid Azure Active Directory join for federated domains] (https://docs.microsoft.com/azure/active-directory/devices/hybrid-azuread-join-federated-domains) to learn more about setting up Azure Active Directory Connect for a simplified join flow for Azure AD device registration.
```

>
> You can learn about this and more by reading [Introduction to Device Management in Azure Active Directory.](https://docs.microsoft.com/azure/active-directory/device-management-introduction)

>[!IMPORTANT]

> > To use Hybrid Identity with Azure Active Directory (HIAAD) and device WriteBack features you must use the in-built GUI with the [latest updates for ADConnect.](https://www.microsoft.com/download/details.aspx?id=47594) 

Reviewer 2: ```suggestion
> To use Hybrid Identity with Azure Active Directory (HIAAD) and device WriteBack features, you must use the built-in GUI with the [latest updates for ADConnect.](https://www.microsoft.com/download/details.aspx?id=47594) 
```

> > To use Hybrid Identity with Azure Active Directory (HIAAD) and device WriteBack features, you must use the built-in GUI with the [latest updates for ADConnect.](https://www.microsoft.com/download/details.aspx?id=47594) 

Reviewer: ```suggestion
> To use Hybrid Identity with Azure Active Directory (HIAAD) and device WriteBack features, you must use the built-in GUI with the [latest updates for ADConnect](https://www.microsoft.com/download/details.aspx?id=47594).
```

Reviewer: You are all too fast for me 😄

>
> You can learn about this and more by reading [Introduction to Device Management in Azure Active Directory.](https://docs.microsoft.com/azure/active-directory/device-management-introduction)

> 

Reviewer 3: Redundant blank line, but mostly cosmetic (there is already a lot of this in existing source).

Author: Done

   `PS C:\> Initialize-ADDeviceRegistration -ServiceAccountName "<your service account>" ` 
3. On the pop-up window click **Yes**.

> 

Reviewer 3: ```suggestion
```
(cosmetic, but still an addition of a redundant blank line)

	@RuleName = "Issue objectSID for domain-joined computers"
	c1:[
		Type == "http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid", 
		Value =~ "-515$", 
		Issuer =~ "^(AD AUTHORITY|SELF AUTHORITY|LOCAL AUTHORITY)$"
	]
	&& 
	c2:[
		Type == "http://schemas.microsoft.com/ws/2008/06/identity/claims/primarysid", 
		Issuer =~ "^(AD AUTHORITY|SELF AUTHORITY|LOCAL AUTHORITY)$"
	]
	=> issue(claim = c2);

#### Issue issuerID for computer when multiple verified domain names in Azure AD

**`http://schemas.microsoft.com/ws/2008/06/identity/claims/issuerid`** - This claim must contain the Uniform Resource Identifier (URI) of any of the verified domain names that connect with the on-premises federation service (AD FS or 3rd party) issuing the token. In AD FS, you can add issuance transform rules that look like the ones below in that specific order after the ones above. Please note that one rule to explicitly issue the rule for users is necessary. In the rules below, a first rule identifying user vs. computer authentication is added.

> 	@RuleName = "Issue account type with the value User when its not a computer"

Reviewer 2: ```suggestion
	@RuleName = "Issue account type with the value User when it's not a computer"
```

**`http://schemas.microsoft.com/ws/2008/06/identity/claims/issuerid`** - This claim must contain the Uniform Resource Identifier (URI) of any of the verified domain names that connect with the on-premises federation service (AD FS or 3rd party) issuing the token. In AD FS, you can add issuance transform rules that look like the ones below in that specific order after the ones above. Please note that one rule to explicitly issue the rule for users is necessary. In the rules below, a first rule identifying user vs. computer authentication is added.

>     @RuleName = "Issue account type with the value User when its not a computer"

Reviewer 3: ```suggestion
    @RuleName = "Issue account type with the value User when it is not a computer"
```

**`http://schemas.microsoft.com/ws/2008/06/identity/claims/issuerid`** - This claim must contain the Uniform Resource Identifier (URI) of any of the verified domain names that connect with the on-premises federation service (AD FS or 3rd party) issuing the token. In AD FS, you can add issuance transform rules that look like the ones below in that specific order after the ones above. Please note that one rule to explicitly issue the rule for users is necessary. In the rules below, a first rule identifying user vs. computer authentication is added.

    @RuleName = "Issue account type with the value User when its not a computer"

> 

Reviewer 3: I don't know if it is necessary to add a blank line here.
Do you know if this blank line is needed as a separator here?

Author: It's not required, but it's code.
On code, it's always good to leave some blank lines for readability

        Type = "http://schemas.microsoft.com/ws/2012/01/accounttype", 
        Value = "DJ"

>     );'

Reviewer 3: I did not notice this earlier, but now it looks obvious to me.
I think we should keep these blank lines in the code samples,
not because of functionality, but for _user readability_.
(Programming codestyle often dictates blank lines between functions/hooks.)

Author: I think you modified another version of the files, not the last ones
I don't see extra lines in those lines in my PR
![image](https://user-images.githubusercontent.com/5489996/58686203-c99bf700-8343-11e9-9196-d262cca64b6f.png)

        Type = "http://schemas.microsoft.com/ws/2012/01/accounttype", 
        Value = "DJ"
    );'

>     

Reviewer 3: ```suggestion

```

        query = ";objectguid;{0}", 
        param = c2.Value

>     );'

Reviewer 3: (keep this blank line for readability/codestyle)

        query = ";objectguid;{0}", 
        param = c2.Value
    );'

>     

Reviewer 3: ```suggestion

```

        Issuer =~ "^(AD AUTHORITY|SELF AUTHORITY|LOCAL AUTHORITY)$"
    ]

>     => issue(claim = c2);'

Reviewer 3: (keep this blank line for readability/codestyle)

        Issuer =~ "^(AD AUTHORITY|SELF AUTHORITY|LOCAL AUTHORITY)$"
    ]
    => issue(claim = c2);'

>     

Reviewer 3: ```suggestion

```

        Value = "http://' + $oneOfVerifiedDomainNames + '/adfs/services/trust/"
    );'

>     }

Reviewer 3: (keep this blank line for readability/codestyle)

        Value = "http://' + $oneOfVerifiedDomainNames + '/adfs/services/trust/"
    );'
    }

>     

Reviewer 3: ```suggestion

```

        param = c2.Value
    );'

>     }

Reviewer 3: (keep this blank line for readability/codestyle)

        param = c2.Value
    );'
    }

>     

Reviewer 3: ```suggestion

```

    $existingRules = (Get-ADFSRelyingPartyTrust -Identifier urn:federation:MicrosoftOnline).IssuanceTransformRules 

>     

Reviewer 3: ```suggestion

```

    $updatedRules = $existingRules + $rule1 + $rule2 + $rule3 + $rule4 + $rule5

>     

Reviewer 3: ```suggestion

```

    $crSet = New-ADFSClaimRuleSet -ClaimRule $updatedRules 

>     

Reviewer 3: ```suggestion

```

For your reference, below is a comprehensive list of the AD DS devices, containers and permissions required for device write-back and authentication to work

- object of type ms-DS-DeviceContainer at CN=RegisteredDevices,DC=&lt;domain&gt;  		
    - read access to the AD FS service account   
    - read/write access to the Azure AD Connect sync AD connector account
- Container CN=Device Registration Configuration,CN=Services,CN=Configuration,DC=&lt;domain&gt;
- Container Device Registration Service DKM under the above container

![Device Registration](images/hybridct/device8.png) 
 
- object of type serviceConnectionpoint at CN=&lt;guid&gt;, CN=Device Registration Configuration,CN=Services,CN=Configuration,DC=&lt;domain&gt;  

> - read/write access to the specified AD connector account name on the new object 

Reviewer 3: ```suggestion
  - read/write access to the specified AD connector account name on the new object 
```
I think this indentation is intentional, because it follows the same pattern as above the image ("read/write access to the Azure AD Connect sync AD connector account"). It is meant to be a subset of the sentence preceding it.
